test/node/parser/serializer.test.ts

```diff
@@ -14,4 +14,57 @@ describe('serialize()', () => {
       ])
     );
   });
+
+  it('returns an empty document when the root input is nullish', () => {
+    // @ts-expect-error: Testing nullish input, not supported by type defs but code works gracefully
+    const emptyDocumentUndef = BSON.serialize(undefined);
+    expect(emptyDocumentUndef).to.deep.equal(new Uint8Array([5, 0, 0, 0, 0]));
+    // @ts-expect-error: Testing nullish input, not supported by type defs but code works gracefully
+    const emptyDocumentNull = BSON.serialize(null);
+    expect(emptyDocumentNull).to.deep.equal(new Uint8Array([5, 0, 0, 0, 0]));
+  });
+
+  it('does not turn nested nulls into empty documents', () => {
+    const nestedNull = bufferFromHexArray([
+      '0A', // null type
+      '6100', // 'a\x00'
+      '' // null is encoded as nothing
+    ]);
+    const emptyDocumentUndef = BSON.serialize({ a: undefined }, { ignoreUndefined: false });
+    expect(emptyDocumentUndef).to.deep.equal(nestedNull);
+    const emptyDocumentNull = BSON.serialize({ a: null });
+    expect(emptyDocumentNull).to.deep.equal(nestedNull);
+  });
+
+  describe('validates input types', () => {
+    it('does not permit arrays as the root input', () => {
+      expect(() => BSON.serialize([])).to.throw(/does not support an array/);
+    });
+
+    it('does not permit objects with a _bsontype string to be serialized at the root', () => {
```

Comment:
Can we add one more test for non-string `_bsontype`?

Reply:
Done, and added more prevention the common types we discussed